test/topology_custom/test_table_desc_read_barrier.py

```diff
@@ -0,0 +1,61 @@
+#
+# Copyright (C) 2024-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+
+import logging
+import time
+import pytest
+
+from test.pylib.manager_client import ManagerClient
+from test.pylib.rest_client import inject_error
+from test.pylib.util import wait_for_cql_and_get_hosts
+from test.topology.conftest import skip_mode
+
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.mark.asyncio
+@skip_mode('release', 'error injections are not supported in release mode')
+async def test_table_desc_read_barrier(request, manager: ManagerClient) -> None:
+    """
+    Regression test for #19213.
+
+    Test that the schema is always updated after read barrier.
+
+    Arrange:
+        - start 2 nodes
+        - create a table
+        - inject error (sleep) on the first node
+        - disable the schema agreement wait
+    Act:
+        - alter the table (change schema) on the second node
+        - trigger the read barrier
+    Assert:
+        - the schema retrieved on both hosts matches
+    """
+    servers = await manager.servers_add(2)
+
+    logging.info("Waiting until driver connects to every server")
+    cql = manager.get_cql()
+    # this is needed so that the hosts will be matching the servers
+    hosts = [(await wait_for_cql_and_get_hosts(cql, [s], time.time() + 60))[0] for s in servers]
```

Comment:
@kbr-scylla is this "wait_for_cql_and_get_hosts" necessary? Can't the test code assume that after it calls servers_add(2) and get_cql(), it actually has access to both nodes in CQL requests?

Reply:
I still need the hosts at some point in the test (and in particular, in the same order as servers, or at least one of them to match) ... although previously I needed both, now I mostly just need the whole list, so perhaps I could use `cql.cluster.metadata.all_hosts()` (not sure about the ordering though)